src/lib.rs

```diff
@@ -6,22 +6,32 @@
 //! They are not necessarily the fastest versions: there are prime sieve methods that
 //! compute the factorial in `O(n (log n loglog n)^2)`. Patches are welcome.
 
+#[cfg(test)]
+#[macro_use]
+extern crate approx;
 #[cfg(test)]
 extern crate num_bigint;
-extern crate num_traits;
 
-use num_traits::{CheckedMul, Unsigned};
+use num_traits::{CheckedMul, Float, FloatConst, Signed, Unsigned};
+
+/// Unary operator for computing the factorial of a number (supertrait)
+pub trait Factorial<Target = Self>: UnsignedFactorial<Target> + SignedFactorial<Target> {}
```

Comment:
Can you even implement this, is there any type for which Unsigned and Signed are both implemented?

Reply:
Nope, doesn't make any sense. Removed.